app/models/products/user_note_mode.rb

```diff
@@ -0,0 +1,74 @@
+module Products
+
+  class UserNoteMode
+
+    VALID_MODES = %w(hidden optional required).freeze
+
+    attr_reader :raw_value
+
+    # An array of UserNoteModes representing the possible modes
+    # Use this in validations and form collections.
+    def self.all
+      VALID_MODES.map { |value| new(value) }
+    end
+
+    def self.from_string(mode_string)
```

Comment:
I like the idempotent version, though I'm not sure that `UserNoteMode.from_string("thing")` is much better than putting the logic in the constructor and having it be `UserNoteMode.new("thing"). I might be inclined to this bit of weirdness:

```
VALID_MODES.each do |mode_string|
  define_singleton_method mode_string.to_sym do
    new(mode_string)
  end
end
```

Which gives you `UserNoteMode.hidden` and the like, but which might be too confusing to use.